### ISSUE TITLE
Not working on LG G5 android 7.0

### ISSUE THREAD

**Issue by LGfrust** [pR0Ps/DisableBatteryWarnings#6]
Is there a reason this isn't working/something I can do to make it work?

version 1.1 suppose to support Nougat, no? I tried both versions, not working... 

LG G5 H830
Android version 7.0
Kernel 3.18.31
Security sofware version
MDF v2.0 Release 4

Xposed 89
Module version 1.1

**Comment by pR0Ps** [comment]
Yes, it should work on 7.0. Do you see any logs from the module?

**Comment by LGfrust** [comment]
I don't see anything in the logs that relates to it, but I'm a beginner. 
Here's the log from today's reboot:
01-02 20:27:03.100 I/Xposed  (  785): -----------------
01-02 20:27:03.100 I/Xposed  (  785): Starting Xposed version 89, compiled for SDK 24
01-02 20:27:03.100 I/Xposed  (  785): Device: LG-H830 (LGE), Android version 7.0 (SDK 24)
01-02 20:27:03.100 I/Xposed  (  785): ROM: NRD90U
01-02 20:27:03.100 I/Xposed  (  785): Build fingerprint: lge/h1_tmo_us/h1:7.0/NRD90U/1631213003394:user/release-keys
01-02 20:27:03.100 I/Xposed  (  785): Platform: arm64-v8a, 64-bit binary, system server: yes
01-02 20:27:03.100 I/Xposed  (  785): SELinux enabled: yes, enforcing: no
01-02 20:27:06.588 I/Xposed  (  785): -----------------
01-02 20:27:06.590 I/Xposed  (  785): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
01-02 20:27:07.515 I/Xposed  (  785): Detected ART runtime
01-02 20:27:07.517 I/Xposed  (  785): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
01-02 20:27:07.742 I/Xposed  (  785): Loading modules from /data/app/eu.faircode.xlua-1/base.apk
01-02 20:27:07.756 I/Xposed  (  785):   Loading class eu.faircode.xlua.XLua
01-02 20:27:07.760 I/Xposed  (  785): Loading modules from /data/app/ru.timjosten.disableipv6-1/base.apk
01-02 20:27:07.768 I/Xposed  (  785):   Loading class ru.timjosten.disableipv6.DisableIPv6
01-02 20:27:07.768 I/Xposed  (  785): Loading modules from /data/app/android.wifiregionalfix-1/base.apk
01-02 20:27:07.775 I/Xposed  (  785):   Loading class android.wifiregionalfix.XMain
01-02 20:27:07.781 I/Xposed  (  785): Loading modules from /data/app/de.defim.apk.lightningwall-1/base.apk
01-02 20:27:07.792 I/Xposed  (  785):   Loading class de.defim.apk.lightningwall.Xposed
01-02 20:27:07.801 I/Xposed  (  785):     [LightningWall] initialized
01-02 20:27:07.801 I/Xposed  (  785): Loading modules from /data/app/ca.cmetcalfe.xposed.disablebatterywarnings-1/base.apk
01-02 20:27:07.812 I/Xposed  (  785):   Loading class ca.cmetcalfe.xposed.disablebatterywarnings.Main
01-02 20:27:13.700 I/Xposed  (  786): -----------------
01-02 20:27:13.700 I/Xposed  (  786): Starting Xposed version 89, compiled for SDK 24
01-02 20:27:13.700 I/Xposed  (  786): Device: LG-H830 (LGE), Android version 7.0 (SDK 24)
01-02 20:27:13.700 I/Xposed  (  786): ROM: NRD90U
01-02 20:27:13.700 I/Xposed  (  786): Build fingerprint: lge/h1_tmo_us/h1:7.0/NRD90U/1631213003394:user/release-keys
01-02 20:27:13.700 I/Xposed  (  786): Platform: arm64-v8a, 32-bit binary, system server: no
01-02 20:27:13.700 I/Xposed  (  786): SELinux enabled: yes, enforcing: no
01-02 20:27:15.738 I/Xposed  (  786): -----------------
01-02 20:27:15.738 I/Xposed  (  786): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
01-02 20:27:16.291 I/Xposed  (  786): Detected ART runtime
01-02 20:27:16.293 I/Xposed  (  786): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
01-02 20:27:16.494 I/Xposed  (  786): Loading modules from /data/app/eu.faircode.xlua-1/base.apk
01-02 20:27:16.509 I/Xposed  (  786):   Loading class eu.faircode.xlua.XLua
01-02 20:27:16.517 I/Xposed  (  786): Loading modules from /data/app/ru.timjosten.disableipv6-1/base.apk
01-02 20:27:16.526 I/Xposed  (  786):   Loading class ru.timjosten.disableipv6.DisableIPv6
01-02 20:27:16.527 I/Xposed  (  786): Loading modules from /data/app/android.wifiregionalfix-1/base.apk
01-02 20:27:16.536 I/Xposed  (  786):   Loading class android.wifiregionalfix.XMain
01-02 20:27:16.541 I/Xposed  (  786): Loading modules from /data/app/de.defim.apk.lightningwall-1/base.apk
01-02 20:27:16.553 I/Xposed  (  786):   Loading class de.defim.apk.lightningwall.Xposed
01-02 20:27:16.564 I/Xposed  (  786):     [LightningWall] initialized
01-02 20:27:16.564 I/Xposed  (  786): Loading modules from /data/app/ca.cmetcalfe.xposed.disablebatterywarnings-1/base.apk
01-02 20:27:16.573 I/Xposed  (  786):   Loading class ca.cmetcalfe.xposed.disablebatterywarnings.Main
01-02 20:27:24.685 W/Xposed  ( 2310): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:25.110 W/Xposed  ( 2408): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:25.350 W/Xposed  ( 2443): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:31.195 W/Xposed  ( 3119): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm64/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm64/system@framework@com.lge.sui.widget@oat@arm64@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:43.961 W/Xposed  ( 4139): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:45.021 W/Xposed  ( 4239): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:49.555 W/Xposed  ( 4519): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:50.059 W/Xposed  ( 4573): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:58.556 W/Xposed  ( 5251): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:27:58.767 W/Xposed  ( 5267): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-02 20:28:47.545 W/Xposed  ( 5570): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory

------
ps - I apologize for the late reply. I was sure no one would reply. Truly appreciate you did!

**Comment by pR0Ps** [comment]
Hmm, I see the module being loaded but I don't see a log saying that the hooks weren't applied.

I assume because your LG G5 runs a modified version of AOSP it's either not loading the `"com.android.systemui"` package (because it's been renamed by LG or something) or the hooks are being applied properly, but the warning is displayed by something else that LG has added so it's not blocking them.

In either of these cases it's likely that I won't be able to fix it, but I've added a log message to verify my suspicions. An update with the new log message should be released soon. Once it does, if you could install it and paste another log here we could maybe see if it can be fixed.

**Comment by LGfrust** [comment]
Just want to make sure - it should drop on the xpose download page, correct?
There's only source here and I checked f-droid as well and there's no apk. (btw, the link to f-droid is incorrect on xposed).
Once the update will be available I'll update reboot and post the log here.

**Comment by pR0Ps** [comment]
Yep, once F-Droid picks up the new version I'll upload it to the xposed repo.

**Comment by LGfrust** [comment]
f-droid took its time.....
Seems like the hooks were applied?
I disabled the other modules so you'll get a cleaner log
Log:
```
01-10 13:29:22.803 I/Xposed  (  871): -----------------
01-10 13:29:22.803 I/Xposed  (  871): Starting Xposed version 89, compiled for SDK 24
01-10 13:29:22.803 I/Xposed  (  871): Device: LG-H830 (LGE), Android version 7.0 (SDK 24)
01-10 13:29:22.803 I/Xposed  (  871): ROM: NRD90U
01-10 13:29:22.803 I/Xposed  (  871): Build fingerprint: lge/h1_tmo_us/h1:7.0/NRD90U/1631213003394:user/release-keys
01-10 13:29:22.803 I/Xposed  (  871): Platform: arm64-v8a, 64-bit binary, system server: yes
01-10 13:29:22.803 I/Xposed  (  871): SELinux enabled: yes, enforcing: no
01-10 13:29:25.440 I/Xposed  (  871): -----------------
01-10 13:29:25.442 I/Xposed  (  871): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
01-10 13:29:26.731 I/Xposed  (  871): Detected ART runtime
01-10 13:29:26.733 I/Xposed  (  871): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
01-10 13:29:26.971 I/Xposed  (  871): Loading modules from /data/app/ca.cmetcalfe.xposed.disablebatterywarnings-2/base.apk
01-10 13:29:26.982 I/Xposed  (  871):   Loading class ca.cmetcalfe.xposed.disablebatterywarnings.Main
01-10 13:29:32.853 I/Xposed  (  872): -----------------
01-10 13:29:32.853 I/Xposed  (  872): Starting Xposed version 89, compiled for SDK 24
01-10 13:29:32.853 I/Xposed  (  872): Device: LG-H830 (LGE), Android version 7.0 (SDK 24)
01-10 13:29:32.853 I/Xposed  (  872): ROM: NRD90U
01-10 13:29:32.853 I/Xposed  (  872): Build fingerprint: lge/h1_tmo_us/h1:7.0/NRD90U/1631213003394:user/release-keys
01-10 13:29:32.853 I/Xposed  (  872): Platform: arm64-v8a, 32-bit binary, system server: no
01-10 13:29:32.853 I/Xposed  (  872): SELinux enabled: yes, enforcing: no
01-10 13:29:34.880 I/Xposed  (  872): -----------------
01-10 13:29:34.880 I/Xposed  (  872): Added Xposed (/system/framework/XposedBridge.jar) to CLASSPATH
01-10 13:29:35.448 I/Xposed  (  872): Detected ART runtime
01-10 13:29:35.451 I/Xposed  (  872): Found Xposed class 'de/robv/android/xposed/XposedBridge', now initializing
01-10 13:29:35.655 I/Xposed  (  872): Loading modules from /data/app/ca.cmetcalfe.xposed.disablebatterywarnings-2/base.apk
01-10 13:29:35.663 I/Xposed  (  872):   Loading class ca.cmetcalfe.xposed.disablebatterywarnings.Main
01-10 13:29:43.609 W/Xposed  ( 2283): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:29:43.716 I/Xposed  ( 2303): DisableBatteryWarnings hooks applied!
01-10 13:29:43.879 W/Xposed  ( 2378): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:29:44.532 W/Xposed  ( 2442): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:29:50.494 W/Xposed  ( 3082): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm64/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm64/system@framework@com.lge.sui.widget@oat@arm64@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:29:58.422 W/Xposed  ( 4004): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:29:59.027 W/Xposed  ( 4095): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:30:03.332 W/Xposed  ( 4399): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:30:04.119 W/Xposed  ( 4463): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:30:10.147 W/Xposed  ( 5167): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
01-10 13:30:10.332 W/Xposed  ( 5188): Failed to load Xposed info for /system/framework/com.lge.sui.widget/oat/arm/com.lge.sui.widget.odex: Unable to open '/data/dalvik-cache/arm/system@framework@com.lge.sui.widget@oat@arm@com.lge.sui.widget.odex@xposed' : No such file or directory
```

Is it a lost cause or can I somehow  get rid of those annoying warnings?

**Comment by pR0Ps** [comment]
If the hooks are being applied and the warning is still showing, it's probably because of something that LG has implemented separately from AOSP. It is possible to find out what that thing is and patch it as well, but would be much more annoying since the source for the modifications is probably not available anywhere.

I don't have the time/desire to do it myself, but I would definitely accept a patch if you or someone else developed and submitted one.
